src/core/internal/cppdemangle.d

```diff
@@ -0,0 +1,311 @@
+/**
+* This module provides functions to demangle C++ symbols
+*
+* For POSIX platforms the function `__cxa_demangle` present in Itanium C++ ABI does the demangling, therefore this module
+* depends on the C++ standard library which must be linked dynamically or available for research through the dynamic loader.
+*
+* For Windows platform the function `UnDecorateSymbolName` present in the Debug Help Library does the demangling,
+* which should be available for all versions of Windows supported by D.
+*
+* Copyright: Copyright © 2020, The D Language Foundation
+* License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+* Authors: Ernesto Castellotti
+* Source: $(DRUNTIMESRC core/internal/_cppdemangle.d)
+* See_Also:
+*      https://itanium-cxx-abi.github.io/cxx-abi/abi.html#demangler
+*      https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-html-USERS-4.3/a01696.html
+*      https://libcxxabi.llvm.org/spec.html
+*      https://docs.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-undecoratesymbolname
+*/
+module core.internal.cppdemangle;
+
+/**
+* Demangles C++ mangled names.
+*
+* The output buffer and return will be contains the demangled name of C++ symbol if the demangling is successful.
+*
+* This function will fail and the mangled name in input will be returned if:
+* 1. The symbol to be processed does not follow the Itanium ABI for POSIX or Visual C++ for Windows, or more simply it is not a C++ name
+* 2. The platform is not compatible or the library needed to perform the demangling was not found or linked with the executable
+* 3. The size of the output buffer is not sufficient to contain the demangled name with its prefix
+*
+* This function will fail and null will be returned if:
+* 1. The output buffer is null
+* 2. The function has failed (see list above) and output buffer is unable to contain the mangled name in input
+*
+ * Params:
+ *  mangledName = The string to demangle.
+ *  outputBuffer = The destination buffer, if the size of the destination buffer is <= the size of cppdemangle output the return string would be incomplete.
+ *  prefix = Specifies the prefix to be added to the beginning of the string containing the demangled name
+ *
+ * Returns:
+ *  The demangled name or the original string if the name is not a mangled C++ name.
+ */
+char[] cppdemangle(const(char)[] mangledName, char[] outputBuffer, string prefix = "") @trusted
```

Comment:
Don't think this function should accept a prefix, it's an unrelated complication and a user can always do that manually:
```
buffer[0..prefix.length] = prefix;
cppdemangle(mangling, buffer[prefix.length .. $]);